tests/unittests/algo/test_evolution_es.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tests for :mod:`orion.algo.evolution_es`."""
+
+import hashlib
+import importlib
+
+import numpy as np
+import pytest
+
+from orion.algo.evolution_es import BracketEVES, compute_budgets, EvolutionES
+from orion.algo.space import Fidelity, Real, Space
+
+
+@pytest.fixture
+def space():
+    """Create a Space with a real dimension and a fidelity value."""
+    space = Space()
+    space.register(Real('lr', 'uniform', 0, 1))
+    space.register(Fidelity('epoch', 1, 9, 1))
+    return space
+
+
+@pytest.fixture
+def space1():
+    """Create a Space with a real dimension and a fidelity value."""
+    space = Space()
+    space.register(Real('lr', 'uniform', 0, 1))
+    space.register(Real('weight_decay', 'uniform', 0, 1))
+    return space
+
+
+@pytest.fixture
+def budgets():
+    """Return a configuration for a bracket."""
+    return [(30, 4), (30, 5), (30, 6)]
+
+
+@pytest.fixture
+def evolution(space):
+    """Return an instance of EvolutionES."""
+    return EvolutionES(space, repetitions=1)
+
+
+@pytest.fixture
+def bracket(budgets, evolution, space):
+    """Return a `Bracket` instance configured with `b_config`."""
+    return BracketEVES(evolution, budgets, 1, space)
+
+
+@pytest.fixture
+def rung_0():
+    """Create fake points and objectives for rung 0."""
+    points = np.linspace(0, 8, 9)
+    return dict(
+        n_trials=9,
+        resources=1,
+        results={hashlib.md5(str([point]).encode('utf-8')).hexdigest(): (point, (1, point))
+                 for point in points})
+
+
+@pytest.fixture
+def rung_1(rung_0):
+    """Create fake points and objectives for rung 1."""
+    values = map(lambda v: (v[0], (3, v[0])), list(sorted(rung_0['results'].values()))[:3])
+    return dict(
+        n_trials=3,
+        resources=3,
+        results={hashlib.md5(str([value[0]]).encode('utf-8')).hexdigest(): value
+                 for value in values})
+
+
+@pytest.fixture
+def rung_2(rung_1):
+    """Create fake points and objectives for rung 1."""
+    values = map(lambda v: (v[0], (9, v[0])), list(sorted(rung_1['results'].values()))[:1])
+    return dict(
+        n_trials=1,
+        resources=9,
+        results={hashlib.md5(str([value[0]]).encode('utf-8')).hexdigest(): value
+                 for value in values})
+
+
+def test_compute_budgets():
+    """Verify proper computation of budgets on a logarithmic scale"""
+    # Check typical values
+    assert compute_budgets(1, 3, 1, 2, 1) == [[(2, 1), (2, 2), (2, 3)]]
+    assert compute_budgets(1, 4, 2, 4, 2) == [[(4, 1), (4, 2), (4, 4)]]
+
+
+def test_get_mutated_candidates(space1):
```

Comment:
The test should call the method get_candidate directly. Copying the code here makes the maintenance difficult and likely to lead to missed errors as we try to harmonize the code in the test and the one in get_candidate. I understand this method is a bit difficult to test as it does several things. I think extracting the different parts of get_candidate into private methods would make it easier to test each private method separately.

1. [The first part of the function](https://github.com/Epistimio/orion/pull/369/files#diff-8d7c467e92a292572c23c35ff41bc207R181-R191) focus on the creation of the population set, with the red and blue teams. This could be moved into a single function which would return all the necessary objects. 
2. After that there is the [loop for mutations](https://github.com/Epistimio/orion/pull/369/files#diff-8d7c467e92a292572c23c35ff41bc207R197-R206). This could be another private method which would be easier to test. I think the verification for duplicates should be done in 2, but if it is not possible it should be done before the creation of the list of points so that we can test the two processes separately. That would be a third private method, and would make the next one the fourth private method.
3. Finally there is the [creation of the list of points](https://github.com/Epistimio/orion/pull/369/files#diff-8d7c467e92a292572c23c35ff41bc207R213-R235).

Reply:
Done. I split the get_candidates function into three functions, and test them directly in the test cases.